bgpd/bgp_evpn.c

```diff
@@ -5111,6 +5111,36 @@ static void evpn_mpattr_encode_type5(struct stream *s, const struct prefix *p,
 		stream_put3(s, 0);
 }
 
+/*
+ * In cases such as 'no advertise-all-vni' and L2 VNI DELETE, we need to
+ * pop all the VPN routes present in the bgp_zebra_announce FIFO yet to
+ * be processed regardless of VNI is configured or not.
+ *
+ * NOTE: NO need to pop the VPN routes in two cases
+ *  1) In free_vni_entry
+ *     - Called by bgp_free()->bgp_evpn_cleanup().
+ *     - Since bgp_delete is called before bgp_free and we pop all the dest
+ *       pertaining to bgp under delete.
+ *  2) evpn_delete_vni() when user configures "no vni" since the withdraw
+ *     of all routes happen in normal cycle.
+ */
+void bgp_zebra_evpn_pop_items_from_announce_fifo(struct bgpevpn *vpn)
+{
+	struct bgp_dest *dest = NULL;
+	struct bgp_dest *dest_next = NULL;
+
+	if (!vpn)
```

Comment:
In what case can this be NULL? Because cleanup_vni_on_disable() is calling this function with a non-NULL argument which is already dereferenced.

Reply:
a mistake ... removed it now